### PR TITLE
feat(relay): replace prometheus with OTEL metrics

### PR DESCRIPTION
**Pull request by thomaseizinger**
To better take advantage of the OTEL ecosystem, we change our prometheus metrics to OTEL metrics. OTEL metrics are pushed to the agent via the OTEL pipeline set up in https://github.com/firezone/firezone/pull/1995 rather than pulled like prometheus.

This means our `/metrics` endpoint is now gone which we previously (ab)used as a health-check. I've added a dedicated `/healthz` endpoint.

### CODE REVIEW

File: terraform/environments/staging/main.tf [hunk 1]
```diff
@@ -632,7 +632,7 @@ module "relays" {
     unhealthy_threshold = 2
 
     http_health_check = {
-      request_path = "/metrics"
+      request_path = "/health"
```
Comment (AndrewDryga) [inline terraform/environments/staging/main.tf]: The convention is to name this endpoint `/healthz`.

**Comment by thomaseizinger** [comment]
Using this branch, I was able to see the metrics exported via OTEL and show up in a local prometheus instance! :tada:

**Comment by thomaseizinger** [comment]
@AndrewDryga It seems that the metrics collected by our agent don't show up in the dashboard anywhere. Do we still need to connect those agents somehow?

**Comment by AndrewDryga** [comment]
[Docs](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/googlecloudexporter/README.md) say that they should just work but maybe we need to experiment with configuration a bit more.

**Comment by thomaseizinger** [comment]
> [Docs](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/googlecloudexporter/README.md) say that they should just work but maybe we need to experiment with configuration a bit more.

I am setting `service.name` so perhaps this paragraph applies to us: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/googlecloudexporter/README.md#preventing-metric-label-collisions

Do we have logs of the exporter?

**Comment by thomaseizinger** [comment]
I've submitted https://github.com/firezone/firezone/pull/2074, let's see if it helps.

**Pull request merged**